jormungandr/src/explorer/mod.rs

```diff
@@ -328,6 +343,45 @@ impl ExplorerDB {
         .await
     }
 
+    pub(self) async fn set_tip(&self, hash: HeaderHash) -> bool {
+        // the tip changes which means now a block is confirmed (at least after
+        // the initial epoch_stability_depth blocks).
+
+        let block = if let Some(state_ref) = self.multiverse.get_ref(hash).await {
+            let state = state_ref.state();
+            Arc::clone(state.blocks.lookup(&hash).unwrap())
+        } else {
+            return false;
+        };
+
+        if let Some(confirmed_block_chain_length) = block
+            .chain_length()
+            .nth_ancestor(self.blockchain_config.epoch_stability_depth)
+        {
+            debug_assert!(
```

Comment:
Why not just `assert!`? Might be difficult to track problems in production.

Reply:
Well, I just thought debug_assert was paranoid enough, and I wanted to write the invariant in some way at least. I don't really mind using `assert` anyway, but if this actually happens it is most likely a programmer error.